restore.py

```diff
@@ -69,11 +69,20 @@ def restoreFromBackup(backup, progress=lambda x: ()):
                 "to restore please use a version of the installer that matches the backup partition")
 
         # format the restore partition(s):
+        restore_fs_type = diskutil.fs_type_from_device(backup_partition)
         try:
-            util.mkfs(constants.rootfs_type, restore_partition)
+            util.mkfs(restore_fs_type, restore_partition)
         except Exception as e:
             raise RuntimeError("Failed to create root filesystem: %s" % e)
 
+        # format the logs partition if the fs_type is changing
+        logs_partition = partitionDevice(disk, logs_partnum)
+        if restore_fs_type != diskutil.fs_type_from_device(logs_partition):
+            try:
+                util.mkfs(restore_fs_type, logs_partition)
+            except Exception as e:
```

Comment:
We will get generic exception used in static code check ,could we use except OSError as e instead of generic exception.